### PR TITLE
Include raw txs data into zkevm_getBatchByNumber

### LINKED ISSUE #2151: Include `raw_txs_data` on `zkevm_getBatchByNumber`
# Rationale

Why should this feature exist?

Batches have this piece of information. Some times the raw data can be reconstructed using transactions data, but this is not always the case. Keep in mind that the raw data can be anything, including random bytes that do not represent Ethereum transactions

What are the use-cases?

1. Simplify the implementation: by just returning data from a single table. The client can do the work of unpacking the raw data, saving time to access the DB on the server, and reducing the amount of data sent through the open internet
2. Be more accurate (as explained in the previous point)

# Implementation

Do you have ideas regarding the implementation of this feature?

Yes, just return all the data from the batch table

Are you willing to implement this feature?

Yes

### PR DESCRIPTION
**Pull request by arnaubennassar**
Closes #2151

### CODE REVIEW

File: jsonrpc/types/types.go [hunk 1]
```diff
@@ -337,10 +337,27 @@ type Batch struct {
...
+) (*Batch, error) {
+	batchL2Data := batch.BatchL2Data
+	if batchL2Data == nil {
+		batchL2dataFromTxs, err := state.EncodeTransactions(batch.Transactions)
+		if err != nil {
+			return nil, fmt.Errorf("error encoding txs into raw data: %w", err)
+		} else {
+			batchL2Data = batchL2dataFromTxs
+		}
+	}
```
Comment (ARR552) [inline jsonrpc/types/types.go:360]: I think this situation is not possible. I mean if the batchL2Data is empty, it's impossible to have decoded transactions. And I'm not sure about this but wouldn't we get an error if the we try to decode an empty array of txs?

Comment (arnaubennassar) [reply jsonrpc/types/types.go:360]: `batchL2Data` is only stored once the batch is closed. Therefore, while the batch is WIP there are txs but not batchL2Data.

> And I'm not sure about this but wouldn't we get an error if the we try to decode an empty array of txs?
Will try it

Comment (ARR552) [reply jsonrpc/types/types.go:360]: Even when the batch is wip the sequencer stores the batchL2Data. You can check it in the state.Batch table
image

Comment (arnaubennassar) [reply jsonrpc/types/types.go]: You're right, I had a misconception about this, glad you've clarified this to me. Fixed

**Review by ARR552 — APPROVED** [review]

**Review by tclemos — APPROVED** [review]

**Pull request merged**